NDVI_seminat.Rmd

```diff
@@ -0,0 +1,989 @@
+
+# check if there's any 0s or 1s (which beta cannot handle)
+summary(SentinelNDVI.seminat$mean_beta) # ok
+summary(ModisNDVI.seminat$mean_beta) # ok
+summary(LandsatNDVI.seminat$mean_beta) # has 1's
+# replace 1s in Landsat data with 0.9999
+LandsatNDVI.seminat <- LandsatNDVI.seminat %>%
+  mutate(mean_beta = replace(mean_beta, mean_beta == 1, 0.9999))
+
+# check if the three Satellite objects have the same structure (for concatenating them)
+names(SentinelNDVI.seminat)
+names(ModisNDVI.seminat)
+names(LandsatNDVI.seminat)
+# Sentinel and Landsat have each an extra column -> omit them when concatenating further below
+# one column is named slightly differently in the Sentinel data -> rename it
+SentinelNDVI.seminat <- SentinelNDVI.seminat %>%
+  dplyr::rename("system.index" = "system:index")
+
+# check if they have the same geometry
+st_crs(SentinelNDVI.seminat)
+st_crs(ModisNDVI.seminat)
+st_crs(LandsatNDVI.seminat)
+# all good
+
+# add an increment to the year variable to avoid overlapping data being hidden in figures
+SentinelNDVI.seminat$year_jit <- SentinelNDVI.seminat$year + 0.3
+ModisNDVI.seminat$year_jit <- ModisNDVI.seminat$year - 0.3
+LandsatNDVI.seminat$year_jit <- LandsatNDVI.seminat$year
+
+# concatenate the three Satellite objects
+allSatNDVI.seminat <- rbind(
+  SentinelNDVI.seminat[, !names(SentinelNDVI.seminat) %in% "subtype"],
+  ModisNDVI.seminat,
+  LandsatNDVI.seminat[, !names(LandsatNDVI.seminat) %in% "column_label"]
+)
+# add variable for Satellite identity
+allSatNDVI.seminat$Sat <- c(
+  rep("Sentinel", nrow(SentinelNDVI.seminat)),
+  rep("Modis", nrow(ModisNDVI.seminat)),
+  rep("Landsat", nrow(LandsatNDVI.seminat))
+)
+allSatNDVI.seminat$Sat <- factor(allSatNDVI.seminat$Sat, levels = c("Sentinel", "Modis", "Landsat"))
+levels(allSatNDVI.seminat$Sat)
+```
+Now we can plot the time series for each main ecosystem type, showing each satellite time series separately:
+```{r}
+# plot
+allSatNDVI.seminat %>%
+  ggplot(aes(x = year_jit, y = mean, color = Sat)) +
+  geom_point() +
+  facet_wrap(~hovedtype, ncol = 1)
+```
+It is quite obvious that the NDVI values from the three satellites are not quantitatively comparable. They vary both in their placement along the y-axis and in their variance. Modis data are absent from the T40/T41-ecosystem types because of a too large pixel size compared to the NiN-polygon sizes. In the other types, Modis is poorly represented for the same reason.
+
+We can test if there are consistent temporal changes in NDVI by running regressions including year as a covariate.
+
+We first investigate a model testing NDVI as a function of year and main ecosystem type for polygons in good condition, and we do so separately for each satellite.
+```{r, warning=FALSE}
+model.seminat.time.GodTilst.Sent <- glmmTMB(mean_beta ~ year * hovedtype + (1 | id), family = beta_family(), data = SentinelNDVI.seminat[SentinelNDVI.seminat$tilstand == "God", ])
```

Comment:
I will cache these models to speed up redndering

Reply:
ok, thanks